src/NuGet.Clients/NuGet.CommandLine/Commands/SignCommand.cs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Security;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Commands;
+using NuGet.Packaging.Signing;
+using NuGet.Shared;
+
+namespace NuGet.CommandLine
+{
+    [Command(typeof(NuGetCommand), "sign", "SignCommandDescription",
+        MinArgs = 1,
+        MaxArgs = 1,
+        UsageSummaryResourceName = "SignCommandUsageSummary",
+        UsageExampleResourceName = "SignCommandUsageExamples",
+        UsageDescriptionResourceName = "SignCommandUsageDescription")]
+    public class SignCommand : Command
+    {
+        // Default constructor used only for testing, since the Command Default Constructor is protected
+        public SignCommand() : base()
+        {
+        }
+
+        public ISignCommandRunner SignCommandRunner { get; set; }
+
+        // List of possible values - https://github.com/Microsoft/referencesource/blob/master/mscorlib/system/security/cryptography/HashAlgorithmName.cs
+        private static string[] _acceptedHashAlgorithms = { "SHA256", "SHA384", "SHA512" };
```

Comment:
note: this will probably need to come from the helper that verifies this later based on the package signing version, that isn't in yet though